its/ruling/src/test/java/org/sonar/java/it/AutoScanTest.java

```diff
@@ -167,28 +170,49 @@ public void javaCheckTestSources() throws Exception {
     List<IssueDiff> rulesSilenced = newDiffs.stream().filter(IssueDiff::onlyFNs).collect(Collectors.toList());
     LOG.info("{} rules silenced without binaries (only FNs):\n{}", rulesSilenced.size(), IssueDiff.prettyPrint(rulesSilenced));
 
-    // store in a JSON file - serializable
-    Files.writeString(pathFor(TARGET_ACTUAL + DIFF_FILE + ".json"), GSON.toJson(newDiffs));
-    // store in a CSV file - can be easily imported in google sheets
+    // Load known diffs
+    var knownDiffFiles = new ArrayList<Path>();
+    try (var dirStream = Files.newDirectoryStream(pathFor("src/test/resources/autoscan/diffs/"),
+      path -> path.getFileName().toString().startsWith("diff_") && path.toString().endsWith(".json"))) {
+      dirStream.forEach(knownDiffFiles::add);
+    }
+    var knownDiffs = new HashMap<String, IssueDiff>();
+    for (var diffFile : knownDiffFiles) {
+      IssueDiff diff = GSON.fromJson(Files.readString(diffFile), GSON_ISSUE_DIFF_TYPE);
+      knownDiffs.put(diff.ruleKey, diff);
+    }
+
+    // store new unexpected diffs in JSON files - serializable
+    Files.createDirectory(pathFor(TARGET_ACTUAL + "autoscan-diffs/"));
+    for (var newDiff : newDiffs) {
+      if (!newDiff.equals(knownDiffs.get(newDiff.ruleKey))) {
+        Files.writeString(pathFor(TARGET_ACTUAL + "autoscan-diffs/diff_" + newDiff.ruleKey + ".json"), GSON.toJson(newDiff));
+      }
+    }
+    // store all new diffs in a CSV file - as an easy import into a spreadsheet application
     Files.writeString(pathFor(TARGET_ACTUAL + DIFF_FILE + ".csv"), IssueDiff.prettyPrint(newDiffs, newTotal));
 
-    List<IssueDiff> knownDiffs = GSON.fromJson(Files.readString(pathFor("src/test/resources/autoscan/" + DIFF_FILE + ".json")), GSON_LIST_ISSUE_DIFF_TYPE);
-    IssueDiff knownTotal = IssueDiff.total(knownDiffs);
+    IssueDiff knownTotal = IssueDiff.total(knownDiffs.values());
 
     SoftAssertions softly = new SoftAssertions();
-    softly.assertThat(newDiffs).containsExactlyInAnyOrderElementsOf(knownDiffs);
+    softly.assertThat(newDiffs).containsExactlyInAnyOrderElementsOf(knownDiffs.values());
     softly.assertThat(newTotal).isEqualTo(knownTotal);
     softly.assertThat(rulesCausingFPs).hasSize(6);
     softly.assertThat(rulesNotReporting).hasSize(7);
-    softly.assertThat(rulesSilenced).hasSize(74);
```

Comment:
The vast majority of rules relying on semantics will be silenced by the absence of binaries. Nowadays, it seems to me that most of our rules that we write rely on semantics. I don't think having this check is very valuable in addition to the diff data, that already tells us if a rule is silenced.